packages/dds/matrix/src/test/memory/sharedMatrix.spec.ts

```diff
@@ -10,57 +10,168 @@ import {
 	benchmarkMemory,
 	isInPerformanceTestingMode,
 } from "@fluid-tools/benchmark";
+import type { IMatrixConsumer } from "@tiny-calc/nano";
 
-import type { SharedMatrix } from "../../index.js";
+import type { ISharedMatrix } from "../../index.js";
 import { UndoRedoStackManager } from "../undoRedoStackManager.js";
-import { createLocalMatrix } from "../utils.js";
+import { createLocalMatrix, type TestMatrixOptions } from "../utils.js";
 
 /**
  * Note: These benchmarks are designed to closely match the benchmarks in SharedTree.
  * If you modify or add tests here, consider updating the corresponding SharedTree benchmarks as well
  * to ensure consistency and comparability between the two implementations.
  */
 
+// TODOs:
+// - single helper function with before and after hooks for setup and teardown
+// - unify with time measurement tests (in terms of API)
+
+/**
+ * Initializes a SharedMatrix for testing.
+ * @remarks Includes initialization of the undo/redo stack, as well as mock event subscriptions.
+ */
+function createMatrix(options: TestMatrixOptions): {
```

Comment:
Note for reviewers: all of the non-undo/redo benchmarks were updated to use this instead of the inline class definition. Otherwise, each of those tests would need to have been updated to bind the event listeners and undo/redo management.